### DISCUSSION TITLE
Cascading deletes when upgrading from prisma 1

### DISCUSSION THREAD

**Discussion by jwld** [prisma/prisma#2719] [Q&A] [answered]
I have 2 questions regarding this.

#### 1) What happens when you run prisma migrate for the first time?

It seems to be adding on delete cascade to every required relation automatically. But for non-required relations it sets to null. Just want to confirm that this is the case as I can't see information on this in the docs anywhere.

#### 2) What naming convention did prisma 1 use for its constraints?
I'm manually updating the non-required constraints to add cascading delete following an answer from [this question](https://github.com/prisma/prisma/discussions/2149):
> ```sql
> ALTER TABLE public.Post
> DROP CONSTRAINT Post_user_fkey,
> ADD CONSTRAINT Post_user_fkey
>    FOREIGN KEY (user)
>    REFERENCES public.User(user)
>    ON DELETE CASCADE
>    ON UPDATE CASCADE;
> ```

Again mainly just for clarity I wanted to check that prisma always follows this naming convention seen with `Post_user_fkey`?

**Comment by ryands17** [comment, marked as answer]
Hey @jwld :wave: 

> #### 1) What happens when you run prisma migrate for the first time?
> It seems to be adding on delete cascade to every required relation automatically. But for non-required relations it sets to null. Just want to confirm that this is the case as I can't see information on this in the docs anywhere.

Yes you're absolutely correct on this behavior.

> #### 2) What naming convention did prisma 1 use for its constraints?
> I'm manually updating the non-required constraints to add cascading delete following an answer from [this question](https://github.com/prisma/prisma/discussions/2149):
> 
> > ```sql
> > ALTER TABLE public.Post
> > DROP CONSTRAINT Post_user_fkey,
> > ADD CONSTRAINT Post_user_fkey
> >    FOREIGN KEY (user)
> >    REFERENCES public.User(user)
> >    ON DELETE CASCADE
> >    ON UPDATE CASCADE;
> > ```
> 
> Again mainly just for clarity I wanted to check that prisma always follows this naming convention seen with `Post_user_fkey`?

As for this as I have run on a Postgres DB, Prisma uses the format of `tableName_foreignKeyFieldname_fkey`.

Let me know if this anwers your question :)

**Reply by jwld** [reply]
Great thanks for the confirmation :)

**Comment by ludralph** [comment]
Hi @jwld,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
